File: perl/dish-decision.pl
#!/usr/bin/env perl

use strict;
use warnings;
use LWP::UserAgent;
use JSON qw ( decode_json );

print 'What Season is it? (Spring, Summer, Fall, Winter)  ';
chomp(my $season = <>);
print 'How many Guests?  ';
chomp(my $guests = <>);

my $ua = LWP::UserAgent->new;
my $uri = 'https://dmn.camunda.cloud/api/v1/decision/example-dish';
my $inputs = '{"season": { "value": "' . $season .  '", "type": "string"}, "guests": {"value": ' . $guests . ', "type": "integer"}}}';
my $req = HTTP::Request->new('POST', $uri);
$req->header( 'Content-Type' => 'application/json' );
$req->content( $inputs );

my $json = $ua->request($req)->decoded_content;
my $result = decode_json( $json )->{'outputs'}{'dish'}{'values'}[0];
print 'You should have '  . $result . "\n";

File: ruby/dish-decision.rb
require 'net/http'
require 'json'

print "What Season is it? (Spring, Summer, Fall, Winter)  "
season = gets.chomp

print "How many Guests?  "
guests = gets.chomp

uri = URI('https://dmn.camunda.cloud/api/v1/decision/example-dish')
Net::HTTP.start(uri.host, uri.port, :use_ssl => true) do |http|
  request = Net::HTTP::Post.new uri
  request.body = {season: {value: season, type: 'string'}, guests: {value: guests.to_i, type: 'integer'}}.to_json
  request.content_type = 'application/json'

  response = http.request request
  result = JSON.parse(response.body)['outputs']['dish']['values'][0]
  puts "You should have #{result}"
end
